dnsapi/dns_nm.sh

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env sh
+
+########################################################################
+# https://namemaster.de hook script for acme.sh
+#
+# Environment variables:
+#
+#  - $NM_user      (your namemaster.de API username)
+#  - $NM_md5       (your namemaster.de API password_as_md5hash)
+#
+# Author: Thilo Gass <thilo.gass@gmail.com>
+# Git repo: https://github.com/ThiloGa/acme.sh
+
+#-- dns_nm_add() - Add TXT record --------------------------------------
+# Usage: dns_nm_add _acme-challenge.subdomain.domain.com "XyZ123..."
+
+dns_nm_add() {
+  fulldomain=$1
+  txt_value=$2
+  _info "Using DNS-01 namemaster hook"
+
+  NM_user="${NM_user:-$(_readaccountconf_mutable NM_user)}"
+  NM_md5="${NM_md5:-$(_readaccountconf_mutable NM_md5)}"
+  if [ -z "$NM_user" ] || [ -z "$NM_md5" ]; then
+    NM_user=""
+    NM_md5=""
+    _err "No auth details provided. Please set user credentials using the \$NM_user and \$NM_md5 environment variables."
+    return 1
+  fi
+  #save the api user and md5 password to the account conf file.
+  _debug "Save user and hash"
+  _saveaccountconf_mutable NM_user "$NM_user"
+  _saveaccountconf_mutable NM_md5 "$NM_md5"
+
+  zone="$(echo "$fulldomain" | _egrep_o "[^.]+.[^.]+$")"
```

Comment:
are you sure you read the guide ?

how can you process a domain like `www.exmaple.co.uk`

Reply:
> are you sure you read the guide ?

yep... :-)

> how can you process a domain like www.exmaple.co.uk

never without an error.
 
the way descriebed in dev-guide of getting and comparing a zone list doesn't work, because there no zone list command in the api right now.

i requested an api improvement.

Reply:
api changes ready
script ready